### ISSUE TITLE
code for interpretation

### ISSUE THREAD

**Issue by Adasunnylily** [YueYANG1996/LaBo#13]
hi, is there any code for interpretation to inference, for example, given a single photo, its classification results and top concepts related?  The interpretability of the model is achieved by obtaining a set of concepts for each image. Is there specific implementation code available? Is it the method demonstrated in Figure 7 of the paper, where the weights corresponding to concepts in the fully connected (fc) layer are selected?
Thanks a lot!

**Comment by YueYANG1996** [comment]
We didn't have a script for single photo inference, but it should be very easy to write, you just need to initialize the model from checkpoint like [this](https://github.com/YueYANG1996/LaBo/blob/66d32ff2376e97b03883e121cc033f55a40fc3e3/main.py#L255C9-L255C28), and following pytorch-lightning's doc to call the [forward function](https://github.com/YueYANG1996/LaBo/blob/66d32ff2376e97b03883e121cc033f55a40fc3e3/models/asso_opt/asso_opt.py#L90) of the model to get the logits on all classes, then do the argmax to make the prediction.

For getting class2concepts association from the weight matrix, we have a function in the model [here](https://github.com/YueYANG1996/LaBo/blob/66d32ff2376e97b03883e121cc033f55a40fc3e3/models/asso_opt/asso_opt.py#L207).

**Comment by Adasunnylily** [comment]
ok, thanks!
